### DISCUSSION TITLE
Hot reload of mounted configMap

### DISCUSSION THREAD

**Discussion by annaazizova** [quarkusio/quarkus#23133] [Q&A] [answered]
I use external config for Quarkus app (mounted to $PWD/config/application.yml), application overrides configuration during runtime. Is there a feature for hot-reload of the external configuration file (to automatically update configuration on mounted ConfigMap change)? Or which is the best way to implement such case?

**Comment by maxandersen** [comment, marked as answer]
We don't support arbitrary reload of config changes.

we do support updates of credentials using a credentials provider, see https://quarkus.io/guides/credentials-provider

for updating other changes we suggest to use kubernetes to restart your process - kubernetes will take care of rerouting traffic to the new nodes.

what kind of changes is it you want to hot reload?

**Reply by annaazizova** [reply]
I want to reload custom configuration parameters

**Reply by radcortez** [reply]
Hi @annaazizova,

We don't support out-of-the-box configuration reload for a number of reasons. Changing a configuration does not mean that Quarkus will change its behavior. For instance, don't expect Quarkus to change the http.port if the value changes. 

Probably this is most suited for user configuration, but we need to be careful to not give the impression that all configuration is reloadable, so we made a decision to not support it.

As a user, if you wish to do so, you can implement your own `ConfigSource` to support such cases. Please check https://quarkus.io/guides/config-extending-support

**Comment by masayag** [comment]
In our project (parodos) we rely heavily on Quarkus when running serverless workflows.
The application is deployed on k8s cluster (as k8s deployment) and has a configmap mounted to it as a volume with the application.properties.
We're looking for a recommended method to restart the Quarkus application when the configuration is changed.
Ideally, without the need to restart the deployment.

@maxandersen has anything changed in approaching this issue since this thread was initiated? or does the recommendation in it remain?

**Reply by radcortez** [reply]
Quarkus does not support reloading configuration once started.

If the configuration is runtime only, restarting Quarkus is enough to reread the configuration and apply the changes.

If the configuration is build time, then the Quarkus applications require a rebuild to apply the changes.
